project-koku/koku
# [COST-3408] tenantprovider foreignkey replacement AWS

## Jira Ticket

[COST-3408](https://issues.redhat.com/browse/COST-3408)

## Description

* Replace the `api.Provider` Foreign Key with `reporting.TenantAPIProvider` for AWS.
* re-prioritize the relationship finding SQL. These changes will get all the relationships through the `reporting.TenantAPIProvider` foreign key before any other relationships. 

## Testing

1. `git checkout main`
2. `make docker-down; make remove-db`
3. `make docker-up-min-no-build`
4. `make create-test-customer; make load-test-customer-data test_source=AWS`
5. Check the AWS reporting numbers, delete the non org-id source, `make clear-cache`, and check the AWS reporting numbers again to ensure they decreased (remaining costs are associated with the org-id AWS source)
6. `git checkout cost-3408-p3`
7. `make run-migrations`
8. `make clear-cache` and again check the AWS reporting numbers to ensure they have not changed.
9. `make delete-test-customer-data; make clear-testing; make clear-trino-data; make clear-cache`
10. check that ALL sources are removed and that there is no AWS reporting costs.
